File: src/main/scala/dev/vrba/harold/modules/sample/SampleModule.scala
package dev.vrba.harold.modules.sample

import dev.vrba.harold.core.HaroldModule
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.entities.Activity
import org.springframework.stereotype.Service

@Service
class SampleModule extends HaroldModule {
  override def name: String = "Sample module"

  override def register(client: JDA): Unit = {
    // Just so it is obvious that the module gets registered...
    client.getPresence.setActivity(Activity.playing("with the sample module"))
  }
}

File: src/main/kotlin/dev/vrba/harold/modules/sample/SampleModule.kt
package dev.vrba.harold.modules.sample

import dev.vrba.harold.core.HaroldModule
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.entities.Activity
import org.springframework.stereotype.Component

@Component
class SampleModule : HaroldModule {
    override val name = "Sample module"

    override fun register(client: JDA) {
        client.presence.activity = Activity.playing("with the sample module")
    }
}